### ISSUE TITLE
Zoom out all the way, then try zooming back in

### ISSUE THREAD

**Issue by i336** [goessner/canvas-area#2]
I haven't checked the code but I suspect there's no lower bounds checking

**Comment by goessner** [comment]
Cannot verify that. With which browser on what OS did you test that .. ?

**Comment by i336** [comment]
Hi, sorry for the delay. I almost completely forgot about this, very glad I didn't close the tab.

I must admit that I filed this issue after only a superficial look at the system. I've had another look and I see my initial impression was a bit off the mark, but not completely so.

What I initially noticed was that, when I zoomed out to an extent that the grid and predrawn elements disappeared, trying to zoom back in wouldn't make them reappear. Or at least it wouldn't *seem* to make them reappear.

Closer inspection revealed that while zooming out then in does work, it can take quite a bit of scrolling for the grid and elements to reappear if they're very very zoomed out, during which there is no visual feedback. This is what I was initially reacting to.

On the surface you might say that this general-purpose library obviously doesn't want to impose some universal notion of "you're zoomed out all the way now" - but thinking about it, this is a canvas container, and the current zoom capabilities make it possible to zoom out of megapixel+ size canvas elements. I strongly doubt that anybody's going to want to be able to zoom out *that much* in practice.

So, methinks that limiting zoom-out to the exact point where the canvas-area goes completely grey and the grid disappears would probably make for sane defaults - but keeping the zoom bounds configurable would make sense, of course.

Also, methinks a "zoom:" indicator at the bottom of the canvas could be useful
